tests/test_cli.py

```diff
@@ -21,14 +21,18 @@
 nsid = "1"
 nsid_ipv6 = "3"
 anagrpid = "2"
-trtype = "TCP"
 gateway_name = socket.gethostname()
 addr = "127.0.0.1"
 addr_ipv6 = "::1"
 server_addr_ipv6 = "2001:db8::3"
-listener_list = [["-g", gateway_name, "-a", addr, "-s", "5001"], ["-g", gateway_name, "-a", addr,"-s", "5002"]]
+listener_list = [["-g", gateway_name, "-a", addr, "-s", "5001", "-t", "tCp", "-f", "ipV4"], ["-g", gateway_name, "-a", addr, "-s", "5002"]]
 listener_list_no_port = [["-g", gateway_name, "-a", addr]]
-listener_list_ipv6 = [["-g", gateway_name, "-a", addr_ipv6, "-s", "5003"], ["-g", gateway_name, "-a", addr_ipv6, "-s", "5004"]]
+listener_list_fc_trtype = [["-g", gateway_name, "-a", addr, "-s", "5010", "--trtype", "FC"]]
+listener_list_invalid_trtype = [["-g", gateway_name, "-a", addr, "-s", "5011", "--trtype", "JUNK"]]
+listener_list_invalid_adrfam = [["-g", gateway_name, "-a", addr, "-s", "5013", "--adrfam", "JUNK"]]
+listener_list_ib_adrfam = [["-g", gateway_name, "-a", addr, "-s", "5014", "--adrfam", "ib"]]
+listener_list_ipv6 = [["-g", gateway_name, "-a", addr_ipv6, "-s", "5003", "--adrfam", "ipv6"], ["-g", gateway_name, "-a", addr_ipv6, "-s", "5004", "--adrfam", "IPV6"]]
+listener_list_discovery = [["-n", "nqn.2014-08.org.nvmexpress.discovery", "-g", gateway_name, "-a", addr, "-s", "5012"]]
```

Comment:
better to use a variable instead of a string for "nqn.2014-08.org.nvmexpress.discovery"